### PR TITLE
Ignore duplicated attributions

### PR DESCRIPTION
**Pull request by aberenyi**
Ignore duplicated attributions - see [#5297](https://github.com/openlayers/ol3/issues/5297).

### CODE REVIEW

**Comment by probins** [comment]
pls squash all these commits

**Comment by aberenyi** [comment]
done

**Comment by tschaub** [comment]
I'm pulling in changes for the upcoming 3.20 release.  I don't feel comfortable bringing in changes without tests that clearly demonstrate the bug and fix.  There are zero attribution control tests currently, so I recognize that this is a fair bit of work for what might feel like a minor change.  But I'd like to see a test that fails in `master` and passes with this change before merging.  Perhaps someone else could lend a hand - especially if this is a real problem that people are experiencing.

**Comment by ahocevar** [comment]
Working on a test right now.

**Comment by aberenyi** [comment]
Many thanks

**Comment by ahocevar** [comment]
Test added.

File: src/ol/control/attribution.js [hunk 1]
```diff
@@ -182,11 +182,21 @@ ol.control.Attribution.prototype.getSourceAttributions = function(frameState) {
...
         intersectsTileRange = false;
       }
+      var attributionAlreadyAdded = false;
       if (intersectsTileRange) {
         if (sourceAttributionKey in hiddenAttributions) {
           delete hiddenAttributions[sourceAttributionKey];
         }
-        attributions[sourceAttributionKey] = sourceAttribution;
+        for (var existingKey in attributions) {
+          if (attributions.hasOwnProperty(existingKey)) {
```
Comment (tschaub) [inline src/ol/control/attribution.js]: We create the `attributions` object and don't guard against people adding enumerable properties to `Object.prototype` elsewhere, so this `hasOwnProperty()` call is not necessary.

**Comment by tschaub** [comment]
@aberenyi or @ahocevar - can you rebase to remove the 6 extra merge commits?

**Comment by tschaub** [comment]
I rebased these commits and added one more in #6230.  @aberenyi and @ahocevar, let me know if you agree with those changes.

**Comment by tschaub** [comment]
Thanks for the contribution @aberenyi and for the tests @ahocevar.  I've merged #6230.

**Comment by probins** [comment]
any reason why this PR only applies to tile sources? If I add 10 vector sources with the same attribution, it's displayed 10 times.

**Comment by ahocevar** [comment]
The whole attribution handling is unnecessarily complicated. We wouldn't need tile handling at all. All we'd need would be an `attribution` config option on `ol.source.Source`, which accepts a `string` or a `function(extent, resolution)`.

**Comment by probins** [comment]
> The whole attribution handling is unnecessarily complicated

I'd certainly agree with that. :-) A related issue is that if you have several visible sources with attributions, there's nothing to tell you which attribution belongs to which source - you have to specify that in the attribution text. If you have for example an OSM layer with vectors, as in the vector-layer example, the OSM attribution implies that the copyright also applies to the vectors.

**Pull request closed without merging**